docs/userguide.md

```diff
@@ -236,106 +239,128 @@ select count(*) from account;
 |---------------| ------------- |-----|-----|
-|Master web UI		  | 8080  | spark.master.ui.port  or SPARK_MASTER_WEBUI_PORT| The value set by the spark.master.ui.port property takes precedence.  |
-|Worker web UI  |   8081| spark.worker.ui.port or SPARK_WORKER_WEBUI_PORT  | The value set by the spark.worker.ui.port takes precedence.|
-|History server web UI   |18080   | spark.history.ui.port	  |Optional; only applies if you use the history server.   |
-|Master port   | 7077  | 		SPARK_MASTER_PORT	  |   |
-|Master REST port   | 6066  | spark.master.rest.port  | Not needed if you disable the REST service.   |
-|Worker port | 	(random)   |  SPARK_WORKER_PORT |   |
+|Master web UI  | `8080`  | spark.master.ui.port  or SPARK_MASTER_WEBUI_PORT| The value set by `spark.master.ui.port` takes precedence.  |
+|Worker web UI  |  `8081`  | spark.worker.ui.port or SPARK_WORKER_WEBUI_PORT  | The value set by `spark.worker.ui.port` takes precedence.|
+|History server web UI   |  `18080`  | spark.history.ui.port  |Optional; it is only applied if you use the history server.   |
+|Master port   |  `7077`  |   SPARK_MASTER_PORT  |   |
+|Master REST port   |  `6066`  | spark.master.rest.port  | Not needed if you disable the `REST` service.   |
+|Worker port |  (random)   |  SPARK_WORKER_PORT |   |
 |Block manager port  |(random)   | spark.blockManager.port  |   |
-|Shuffle server	  |7337   | spark.shuffle.service.port  | 		Optional; only applies if you use the external shuffle service.  |
-|Application web UI	|4040|	spark.ui.port | if 4040 is used, then 4041 will be used
+|Shuffle server  |  `7337`   | spark.shuffle.service.port  |  Optional; it is only applied if you use the external shuffle service.  |
+|  Application web UI  |  `4040`  |  spark.ui.port | If `4040` has been occupied, then `4041` is used. |
 
-## FAQ
+## 13. FAQ
 
-Q: What are the pros/cons of independent deployment as opposed to a shared resource with an existing Spark / Hadoop cluster?
+Q: What are the pros and cons of independent deployment as opposed to a shared resource with an existing Spark / Hadoop cluster?
 
 A: You can use the existing Spark cluster without a separate deployment, but if the existing cluster is busy, TiSpark will not be able to achieve the desired speed.
 
 Q: Can I mix Spark with TiKV?
 
-A: If TiDB and TiKV are overloaded and run critical online tasks, consider deploying TiSpark separately. You also need to consider using different NICs to ensure that OLTP's network resources are not compromised and affect online business. If the online business requirements are not high or the loading is not large enough, you can consider mixing TiSpark with TiKV deployment.
+A: If TiDB and TiKV are overloaded and run critical online tasks, consider deploying TiSpark separately.
+
+You also need to consider using different NICs to ensure that OLTP's network resources are not compromised and do not affect online business.
```

Comment:
```suggestion
You also need to consider using different NICs to ensure that OLTP's network resources are not compromised so that online business is not affected.
```